docs/tutorials/09_application_classes.ipynb

```diff
@@ -347,13 +345,13 @@
       "\n",
       "solution: [0, 1, 3]\n",
       "\n",
-      "time: 4.045089960098267\n"
+      "time: 33.314785957336426\n"
```

Comment:
Is the 4s to 33s just due to your machine? I see the published one has 3.3s so maybe CI ran the old one faster than the  4s - but 33s seems real slow in comparision. I just downloaded the tutorials 3.8 zip where CI runs these and there its 20s so still quite a lot slower than the 3.3s from before!

Also before the quantum instance was qasm with shots=1000 - this tutorial now computed with an ideal distribution ie Sampler() without specifying any shots. I guess thats fine given this is more about the application classes.

Reply:
I notice that this notebook takes longer than before. I need to investigate more about SamplingVQE.

Reply:
I took a profile and found that `_DiagonalEstimator._evaluate_sparsepuali` is the bottleneck. I will think of a fix for the performance issue.
 https://github.com/Qiskit/qiskit-terra/blob/8bac5d46177756271fbb5330d053bccaf625b729/qiskit/algorithms/minimum_eigensolvers/diagonal_estimator.py#L184-L188
![image](https://user-images.githubusercontent.com/31178928/205242586-08af02e8-8bf2-4c52-aa4c-42db860a7e34.png)

Reply:
I made a fix https://github.com/Qiskit/qiskit-terra/pull/9229. The note gets as fast as previous.
![image](https://user-images.githubusercontent.com/31178928/205251035-33693ffa-894c-4645-9e99-3ac2b155ffe6.png)